File: greedy/partition_array_such_that_max_diff_is_k.java
public int partitionArray(int[] nums, int k)
{
    Arrays.sort(nums);
    
    int ans = 1;
    int x = nums[0];

    for (int i = 1; i < nums.length; i++)
    {
        if (nums[i] - x > k)
        {
            x = nums[i];
            ans++;
        }
    }

    return ans;
}

File: greedy/partition_array_such_that_max_diff_is_k.py
def partitionArray(nums: list[int], k: int) -> int:
    nums.sort()

    ans = 1
    x = nums[0]

    for i in range(1, len(nums)):
        if nums[i] - x > k:
            x = nums[i]
            ans += 1
    
    return ans